### ISSUE TITLE
Samba Clustering - Syntax Error

### ISSUE THREAD

**Issue by copenlc** [imker25/samba_exporter#100]
Hey imker25 sorry if this is out of line, loving what you made by the way.  But ran into this error when experimenting with clustered SMB. 

SMBSTATUS is putting a number : PID and it looks like samba_exporter is having issues processing the data.  Wondering if you had any thoughts ? 

example from smbstatus 

2:42597 nobody       nogroup      10.63.1.55 (ipv4:10.63.1.55:57033)        SMB3_11           -                    -

error from running samba_exporter with a -v flag.  

strconv.Atoi: parsing "1:25648": invalid syntax

If you need any additional information let me know.

**Comment by imker25** [comment]
Hi copenlc,

thanks for this report. To fix this issue I will need indeed some more data.

Please run the following commands and attach the created output files to this issue:

```bash
sudo smbstatus -L > /tmp/smbstatus_L.log
sudo smbstatus -p > /tmp/smbstatus_p.log
sudo smbstatus -s >  /tmp/smbstatus_S.log
```

It would also be good to see the one full output of the error message, so you could maybe post some lines from each of the following commands output: 

```bash
sudo journalctl -b -u samba_statusd.service
sudo journalctl -b -u samba_exporter.service
```

Thank you
imker

**Comment by copenlc** [comment]
Posted some snippets below imker let me know if this helps / you need anything more. 

```
root@test-post-pr-od1:~# sudo journalctl -b -u samba_statusd.service
-- Logs begin at Tue 2023-04-04 13:40:31 PDT, end at Tue 2023-04-04 14:35:49 PDT. --
Apr 04 14:23:48 test-post-pr-od1 systemd[1]: Stopping samba-statusd service for prometheus samba-exporter, based on https://github.com/imker25/samba_exporter...
Apr 04 14:23:48 test-post-pr-od1 systemd[1]: samba_statusd.service: Main process exited, code=killed, status=15/TERM
Apr 04 14:23:48 test-post-pr-od1 systemd[1]: samba_statusd.service: Succeeded.
Apr 04 14:23:48 test-post-pr-od1 systemd[1]: Stopped samba-statusd service for prometheus samba-exporter, based on https://github.com/imker25/samba_exporter.
Apr 04 14:32:21 test-post-pr-od1 systemd[1]: Starting samba-statusd service for prometheus samba-exporter, based on https://github.com/imker25/samba_exporter...
Apr 04 14:32:21 test-post-pr-od1 start_samba_statusd[25225]: Information: Started /usr/bin/samba_statusd, waiting for requests in pipe
Apr 04 14:32:21 test-post-pr-od1 systemd[1]: Started samba-statusd service for prometheus samba-exporter, based on https://github.com/imker25/samba_exporter.
root@test-post-pr-od1:~# sudo journalctl -b -u samba_exporter.service
-- Logs begin at Tue 2023-04-04 13:40:31 PDT, end at Tue 2023-04-04 14:35:52 PDT. --
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:25648": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:29723": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:25648": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:36092": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:25648": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:57086": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:55399": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:36092": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:25648": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:57086": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:55399": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:36092": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:25648": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:29723": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:55399": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:57086": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "1:25648": invalid syntax
Apr 04 13:40:31 test-post-pr-od1 samba_exporter[47393]: strconv.Atoi: parsing "3:57086": invalid syntax

smbstatus -L 

Locked files:
Pid          Uid        DenyMode   Access      R/W        Oplock           SharePath   Name   Time
--------------------------------------------------------------------------------------------------
1:55399      1001       DENY_NONE  0x12019f    RDWR       LEASE(RWH)       /lfsmnt/dst01   test-post-prod-od/testI Special/test_I_E_L/testI Special/Clip/792_2134.MXF 48000_11.pek   Tue Apr  4 14:23:18 2023
1:55399      1001       DENY_WRITE 0x120089    RDONLY     LEASE(RWH)       /lfsmnt/dst01   test-post-prod-od/testI Special/test_I_E_L/testI Special/Clip/Clip0005.MXF 48000_21.pek   Tue Apr  4 14:26:09 2023
1:55399      1001       DENY_WRITE 0x120089    RDONLY     LEASE(RWH)       /lfsmnt/dst01   test-post-prod-od/testI Special/test_I_E_L/testI Special/NEMOS GARDEN/drone/DCIM/100MEDIA/DJI_0177.MOV   Tue Apr  4 14:32:01 2023
1:55399      1001       DENY_NONE  0x120089    RDONLY     LEASE(RWH)       /lfsmnt/dst01   test-post-prod-od/testI Special/test_I_E_L/testI Special/NEMOS GARDEN/drone/DCIM/100MEDIA/DJI_0177.MOV   Tue Apr  4 14:32:01 2023
1:19801      1001       DENY_NONE  0x100081    RDONLY     NONE             /lfsmnt/dst01   test-post-prod-od/OAP/Presentations/test_The_Whole.mov  Tue Apr  4 03:17:50 2023
3:57086      1001       DENY_NONE  0x120089    RDONLY     LEASE(RWH)       /lfsmnt/dst01   test-post-prod-od/Day 8/B camera/CLIPS001/CC0639/CC063904.MXF   Tue Apr  4 08:11:39 2023
1:55399      1001       DENY_WRITE 0x120089    RDONLY     LEASE(RWH)       /lfsmnt/dst01   test-post-prod-od/test.wav 48000.pek   Tue Apr  4 14:13:28 2023

smbstatus -p

Samba version 4.9.5-Debian
PID     Username     Group        Machine                                   Protocol Version  Encryption           Signing
----------------------------------------------------------------------------------------------------------------------------------------
3:57086 nobody       nogroup      10.63.0.41 (ipv4:10.63.0.41:62834)        SMB3_11           -                    -
3:24179 nobody       nogroup      10.63.0.28 (ipv4:10.63.0.28:58968)        SMB3_11           -                    -
1:19801 nobody       nogroup      10.63.0.36 (ipv4:10.63.0.36:53407)        SMB3_11           -                    -
3:24179 nobody       nogroup      10.63.0.28 (ipv4:10.63.0.28:58968)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -

smbstatus -s 

Samba version 4.9.5-Debian
PID     Username     Group        Machine                                   Protocol Version  Encryption           Signing
----------------------------------------------------------------------------------------------------------------------------------------
1:19801 nobody       nogroup      10.63.0.36 (ipv4:10.63.0.36:53407)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
1:19801 nobody       nogroup      10.63.0.36 (ipv4:10.63.0.36:53407)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
1:25648 nobody       nogroup      10.63.0.81 (ipv4:10.63.0.81:49591)        SMB3_11           -                    -
1:25648 nobody       nogroup      10.63.0.81 (ipv4:10.63.0.81:49591)        SMB3_11           -                    -
1:25648 nobody       nogroup      10.63.0.81 (ipv4:10.63.0.81:49591)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
1:19801 nobody       nogroup      10.63.0.36 (ipv4:10.63.0.36:53407)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
1:55399 nobody       nogroup      10.63.0.11 (ipv4:10.63.0.11:50370)        SMB3_11           -                    -
2:42597 nobody       nogroup      10.63.1.55 (ipv4:10.63.1.55:57033)        SMB3_11           -                    -
1:19801 nobody       nogroup      10.63.0.36 (ipv4:10.63.0.36:53407)        SMB3_11           -                    -
2:42597 nobody       nogroup      10.63.1.55 (ipv4:10.63.1.55:57033)        SMB3_11           -                    -
```

**Comment by imker25** [comment]
Thanks for the data. I think with this I can provided an updated version

**Comment by imker25** [comment]
@copenlc One more question. Do you know what this number in front of the ':' actually means?

**Comment by copenlc** [comment]
That is the node number in the cluster.  In this use case, I am running a 4 node cluster and it starts at 0: for node 1 and going to 3: for node 4.

**Comment by imker25** [comment]
Well, that sounds reasonable. 

I will do the following:
- Fixing this bug, so samba-exporter can handle this cluster style output
- I will add the node number as a exported metric later. This will be tracked in #101

**Comment by imker25** [comment]
Should be fixed with [V1.27.2-pre](https://github.com/imker25/samba_exporter/releases/tag/1.27.2-pre)

@copenlc Can you might install this version and test it?

**Comment by copenlc** [comment]
Hey @imker25 looks like were close only getting that error for lock requests now.  Output below. 

```
root@test-post-pr-od1:~# samba_exporter -verbose
Call:  samba_exporter -verbose
Version: 1.27.2-pre~ppa1~debian11
Verbose: Named pipe for requests: /run/samba_exporter.request.pipe
Verbose: Named pipe for response: /run/samba_exporter.response.pipe
Verbose: Setup prometheus exporter
Verbose: Request samba_statusd to get prometheus descriptions
Verbose: Send "PROCESS_REQUEST:" request with ID 1 on pipe
Verbose: Wait for "PROCESS_REQUEST:" response with ID 1 on pipe
Verbose: Handle "PROCESS_REQUEST:" response with ID 1 from pipe
Verbose: Send "SHARE_REQUEST:" request with ID 2 on pipe
Verbose: Wait for "SHARE_REQUEST:" response with ID 2 on pipe
Verbose: Handle "SHARE_REQUEST:" response with ID 2 from pipe
Verbose: Send "LOCK_REQUEST:" request with ID 3 on pipe
Verbose: Wait for "LOCK_REQUEST:" response with ID 3 on pipe
strconv.Atoi: parsing "0:34520": invalid syntax
strconv.Atoi: parsing "1:70753": invalid syntax
strconv.Atoi: parsing "0:19497": invalid syntax
strconv.Atoi: parsing "2:67078": invalid syntax
strconv.Atoi: parsing "3:3478": invalid syntax
strconv.Atoi: parsing "3:66526": invalid syntax
strconv.Atoi: parsing "0:20781": invalid syntax
strconv.Atoi: parsing "0:34520": invalid syntax
strconv.Atoi: parsing "0:20781": invalid syntax
strconv.Atoi: parsing "3:48181": invalid syntax
strconv.Atoi: parsing "0:19497": invalid syntax
strconv.Atoi: parsing "2:20977": invalid syntax
strconv.Atoi: parsing "3:22112": invalid syntax
strconv.Atoi: parsing "2:67078": invalid syntax
strconv.Atoi: parsing "3:66526": invalid syntax
strconv.Atoi: parsing "3:31474": invalid syntax
strconv.Atoi: parsing "3:31474": invalid syntax
strconv.Atoi: parsing "3:48861": invalid syntax
strconv.Atoi: parsing "2:33385": invalid syntax
strconv.Atoi: parsing "2:33385": invalid syntax
Verbose: Handle "LOCK_REQUEST:" response with ID 3 from pipe
Verbose: Send "PS_REQUEST:" request with ID 4 on pipe
Verbose: Wait for "PS_REQUEST:" response with ID 4 on pipe
Verbose: Handle "PS_REQUEST:" response with ID 4 from pipe
Verbose: Got an empty share table when requesting "smbstatus -S -n" from samba_statusd
Verbose: Handle samba_statusd response and set prometheus descriptions
Information: Started samba_exporter, get metrics on http://:9922/metrics
Verbose: Request samba_statusd to get prometheus metrics
Verbose: Send "PROCESS_REQUEST:" request with ID 5 on pipe
Verbose: Wait for "PROCESS_REQUEST:" response with ID 5 on pipe
Verbose: Handle "PROCESS_REQUEST:" response with ID 5 from pipe
Verbose: Send "SHARE_REQUEST:" request with ID 6 on pipe
Verbose: Wait for "SHARE_REQUEST:" response with ID 6 on pipe
Verbose: Handle "SHARE_REQUEST:" response with ID 6 from pipe
Verbose: Send "LOCK_REQUEST:" request with ID 7 on pipe
Verbose: Wait for "LOCK_REQUEST:" response with ID 7 on pipe
strconv.Atoi: parsing "0:34520": invalid syntax
strconv.Atoi: parsing "1:70753": invalid syntax
strconv.Atoi: parsing "0:19497": invalid syntax
strconv.Atoi: parsing "2:67078": invalid syntax
strconv.Atoi: parsing "3:3478": invalid syntax
strconv.Atoi: parsing "3:66526": invalid syntax
strconv.Atoi: parsing "0:20781": invalid syntax
strconv.Atoi: parsing "3:48181": invalid syntax
strconv.Atoi: parsing "0:34520": invalid syntax
strconv.Atoi: parsing "0:20781": invalid syntax
strconv.Atoi: parsing "3:22112": invalid syntax
strconv.Atoi: parsing "0:19497": invalid syntax
strconv.Atoi: parsing "3:66526": invalid syntax
strconv.Atoi: parsing "3:31474": invalid syntax
strconv.Atoi: parsing "3:31474": invalid syntax
strconv.Atoi: parsing "2:20977": invalid syntax
strconv.Atoi: parsing "3:48861": invalid syntax
strconv.Atoi: parsing "2:67078": invalid syntax
strconv.Atoi: parsing "2:33385": invalid syntax
strconv.Atoi: parsing "2:33385": invalid syntax
Verbose: Handle "LOCK_REQUEST:" response with ID 7 from pipe
Verbose: Send "PS_REQUEST:" request with ID 8 on pipe
Verbose: Wait for "PS_REQUEST:" response with ID 8 on pipe
Verbose: Handle "PS_REQUEST:" response with ID 8 from pipe
Verbose: Got an empty share table when requesting "smbstatus -S -n" from samba_statusd
Verbose: Handle samba_statusd response and set prometheus metrics
Verbose: Request samba_statusd to get prometheus metrics
```

**Comment by imker25** [comment]
@copenlc Can you please test again with [V1.26.7](https://github.com/imker25/samba_exporter/releases/tag/1.26.7). I'm not sure if the issue is fixed now, but the enhanced logging I build in should hopefully give me hints why it is still failing.

Thank you

**Comment by copenlc** [comment]
Hey @imker25 just got V1.27.3-pre installed and looks like the errors are gone =).  Going to try and install on the rest of the cluster and test for a little while and report back if I get any errors.

**Comment by copenlc** [comment]
Looks good so far after a day of testing.  Occasionally getting some drop outs in metric data, but assuming thats cause by my timeout settings running the service. 
![image](https://user-images.githubusercontent.com/25438177/232081662-de1f4a72-5729-499f-84c9-a80b519b2003.png)

**Comment by imker25** [comment]
Well this could be. If it is really caused by timeouts you should see something in the logs. So you maybe want to try a `sudo journalctl -u samba_exporter.service` with the `--since` so you are near this drops. 

This might helps to figure out if you should increase the timeout value.

**Comment by imker25** [comment]
Fixed with [V1.27.5](https://github.com/imker25/samba_exporter/releases/tag/1.27.5)
